### PR TITLE
Jingjli/2023 05 01

### PR DESCRIPTION
**Pull request by jingjlii**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/batch/Microsoft.Azure.Batch/Tools/ObjectModelCodeGenerator/Spec/VMExtension.json [hunk 1]
```diff
@@ -63,6 +63,17 @@
       },
       "Value": null
     },
+    {
+      "Key": {
+        "Type": "bool?",
+        "Name": "EnableAutomaticUpgrade",
+        "SummaryComment": "Indicates whether the extension should be automatically upgraded by the platform if there is a newer version of the extension available.",
+        "RemarksComment": null,
+        "BoundAccess": "read,write",
```
Comment (dpwatrous) [inline sdk/batch/Microsoft.Azure.Batch/Tools/ObjectModelCodeGenerator/Spec/VMExtension.json:72]: Should this be read-only for BoundAccess? I didn't think this was updatable.

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/AddTaskCollectionIntegrationTests.cs [hunk 2]
```diff
@@ -369,7 +369,7 @@ public async Task Bug1360227_AddTasksBatchWithFilesToStage(bool useJobOperations
                 Assert.Equal(expectedInitialFileStagingArtifactsCount, legArtifactsCountList.First());
                 Assert.Equal(expectedFinalFileStagingArtifactsCount, legArtifactsCountList.Last());
             },
-            TestTimeout);
+            LongTestTimeout);
```
Comment (paterasMSFT) [inline sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/AddTaskCollectionIntegrationTests.cs:372]: Don't forget to update the attribute to LongDuration, to reflect this change.

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/CloudPoolIntegrationTests.cs [hunk 3]
```diff
@@ -23,12 +23,11 @@ namespace BatchClientIntegrationTests
     using Xunit.Abstractions;
     using Protocol = Microsoft.Azure.Batch.Protocol;
     using Microsoft.Azure.Batch.Integration.Tests.IntegrationTestUtilities;
-    using System.Net.PeerToPeer;
 
     public class CloudPoolIntegrationTests
     {
         private readonly ITestOutputHelper testOutputHelper;
-        private static readonly TimeSpan LongTestTimeout = TimeSpan.FromMinutes(5);
+        private static readonly TimeSpan LongTestTimeout = TimeSpan.FromMinutes(15);
```
Comment (paterasMSFT) [inline sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/CloudPoolIntegrationTests.cs:30]: Why this increase?

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/CloudPoolIntegrationTests.cs [hunk 4]
```diff
@@ -1139,49 +1138,170 @@ void test()
...
+                    Assert.True(pool.NetworkConfiguration.EnableAcceleratedNetworking);
+                }
+                finally
+                {
+                    //Delete the pool
+                    TestUtilities.DeletePoolIfExistsAsync(batchCli, poolId).Wait();
+                }
+            }
+
+            SynchronizationContextHelper.RunTest(test, LongTestTimeout);
```
Comment (paterasMSFT) [inline sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/CloudPoolIntegrationTests.cs:1240]: Update duration attribute here, too.

File: sdk/batch/Microsoft.Azure.Batch/Tools/ObjectModelCodeGenerator/Spec/VMExtension.json (hunk 1 above)
Comment (jingjlii) [reply sdk/batch/Microsoft.Azure.Batch/Tools/ObjectModelCodeGenerator/Spec/VMExtension.json:72]: Yes, this is not an updatable property, we should change it to read-only for BoundAccess. And I am also considering if we should change all properties in VMExtension model to read-only for BoundAccess...

Comment (jingjlii) [reply sdk/batch/Microsoft.Azure.Batch/Tools/ObjectModelCodeGenerator/Spec/VMExtension.json:72]: The UpdatePool API doesn't update VMExtension, set write for BoundAccess is useless...

File: sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj [hunk 5]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <Description>This client library provides access to the Microsoft Azure Batch service.</Description>
     <Version>15.5.0-beta.1</Version>
-    <ApiCompatVersion>15.4.0</ApiCompatVersion>
+    <!--<ApiCompatVersion>15.4.0</ApiCompatVersion>-->
     <!--The ApiCompatVersion is managed automatically and should not generally be modified manually.-->
```
Comment (jingjlii) [inline sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj:6]: @dpwatrous Should I revert this change? I comment out the ApiCompatVersion to pass build, but seems I shouldn't make this change.

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/AddTaskCollectionIntegrationTests.cs (hunk 2 above)
Comment (jingjlii) [reply sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/AddTaskCollectionIntegrationTests.cs:372]: Thanks, updated the Trait attribute to LongLongDuration as the average test time in my local env is ~5mins

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/CloudPoolIntegrationTests.cs [hunk 6]
```diff
@@ -1139,49 +1138,170 @@ void test()
...
+                finally
+                {
+                    await TestUtilities.DeletePoolIfExistsAsync(batchCli, poolId).ConfigureAwait(false);
+                }
+            }
 
-                    //Then to Classic
-                    pool.TargetNodeCommunicationMode = NodeCommunicationMode.Classic;
-                    await pool.CommitChangesAsync().ConfigureAwait(false);
+            await SynchronizationContextHelper.RunTestAsync(test, LongTestTimeout);
```
[the remark this answers is not in the document]
Comment (jingjlii) [reply sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/CloudPoolIntegrationTests.cs:1202]: Updated as suggestion.

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/CloudPoolIntegrationTests.cs (hunk 4 above)
Comment (jingjlii) [reply sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/CloudPoolIntegrationTests.cs]: Updated as suggestion.

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/CloudPoolIntegrationTests.cs (hunk 3 above)
Comment (jingjlii) [reply sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/CloudPoolIntegrationTests.cs]: I shouldn't wait for 1 min in each pool state check, revert to 5 mins in LongTestTimeout and did some test code changes.

**Review by paterasMSFT — APPROVED** [review]

File: sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj (hunk 5 above)
Comment (jingjlii) [reply sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj]: Reverted.
